crates/trie/trie/src/state.rs

```diff
@@ -252,6 +254,21 @@ impl HashedPostState {
             .with_prefix_sets(prefix_sets)
             .root_with_updates()
     }
+
+    /// Generates the state proof for target account and slots on top of this [`HashedPostState`].
+    pub fn account_proof<TX: DbTx>(
+        &self,
+        tx: &TX,
+        address: Address,
+        slots: &[B256],
+    ) -> Result<AccountProof, StateRootError> {
+        let sorted = self.clone().into_sorted();
```

Comment:
this clones two maps for the into_sorted call that also transforms the maps,
maybe this could be optimized to get rid of this clone and operate on into_iter, but can look at this separately